contracts/0.8.25/utils/V3TemporaryAdmin.sol

```diff
@@ -0,0 +1,201 @@
+     * @dev This is the main external function that should be called after deployment
+     * @param _vaultHub The VaultHub contract address
+     * @param _predepositGuarantee The PredepositGuarantee contract address
+     * @param _lazyOracle The LazyOracle contract address
+     * @param _operatorGrid The OperatorGrid contract address
+     * @param _burner The Burner contract address
+     * @param _csmAccounting The CSM Accounting contract address
+     * @param _codehash The allowed codehash to set on VaultHub
+     */
+    function completeSetup(
+        address _vaultHub,
+        address _predepositGuarantee,
+        address _lazyOracle,
+        address _operatorGrid,
+        address _burner,
+        address _accounting,
+        address _csmAccounting,
+        bytes32 _codehash
+    ) external {
+        if (_vaultHub == address(0)) revert ZeroVaultHub();
+        if (_predepositGuarantee == address(0)) revert ZeroPredepositGuarantee();
+        if (_lazyOracle == address(0)) revert ZeroLazyOracle();
+        if (_operatorGrid == address(0)) revert ZeroOperatorGrid();
+        if (_burner == address(0)) revert ZeroBurner();
+        if (_accounting == address(0)) revert ZeroAccounting();
+        if (_csmAccounting == address(0)) revert ZeroCsmAccounting();
+
+        // Setup VaultHub with all roles and transfer admin
+        _setupVaultHub(_vaultHub, _codehash);
+
+        // Setup PredepositGuarantee and transfer admin
+        _setupPredepositGuarantee(_predepositGuarantee);
+
+        // Setup LazyOracle and transfer admin
+        _setupLazyOracle(_lazyOracle);
+
+        // Setup OperatorGrid and transfer admin
+        _setupOperatorGrid(_operatorGrid);
+
+        // Setup Burner and transfer admin
+        _setupBurner(_burner, _accounting, _csmAccounting);
+    }
+
+
+    /**
+     * @notice Setup VaultHub with all required roles and transfer admin to agent
+     * @param _vaultHub The VaultHub contract address
+     */
+    function _setupVaultHub(address _vaultHub, bytes32 _codehash) private {
+        // Get roles from the contract
+        bytes32 pauseRole = IPausableUntil(_vaultHub).PAUSE_ROLE();
+        bytes32 vaultMasterRole = IVaultHub(_vaultHub).VAULT_MASTER_ROLE();
+        bytes32 vaultCodehashSetRole = IVaultHub(_vaultHub).VAULT_CODEHASH_SET_ROLE();
+        bytes32 redemptionMasterRole = IVaultHub(_vaultHub).REDEMPTION_MASTER_ROLE();
+        bytes32 validatorExitRole = IVaultHub(_vaultHub).VALIDATOR_EXIT_ROLE();
+
+        IAccessControl(_vaultHub).grantRole(vaultCodehashSetRole, address(this));
+        IVaultHub(_vaultHub).setAllowedCodehash(_codehash, true);
+        IAccessControl(_vaultHub).revokeRole(vaultCodehashSetRole, address(this));
+
+        // Grant PAUSE_ROLE to gateSeal
+        IAccessControl(_vaultHub).grantRole(pauseRole, GATE_SEAL);
+
+        // Grant other roles to agent
+        // TODO: does agent need all these roles?
```

Comment:
Need to define this section in detais what goes to agent and what will go to ET

Reply:
the q I still have here is if agent should has any of the roles